terraform/test-fixtures/plan-count-inc-index-list/main.tf

```diff
@@ -0,0 +1,24 @@
+variable "count" {
+    default = 2
+}
+
+resource "aws_instance" "foo" {
+    count = "${var.count}"
+    foo = "foo"
+}
+
+resource "aws_instance" "bar" {
+    count = "${var.count}"
+    foo = "${element(aws_instance.foo.*.foo,count.index)}-bar"
```

Comment:
But if this is changed to `foo = "${element(aws_instance.foo.*.id,count.index)}-bar"`, then it _will_ be marked for update.